sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/handler/ReceiverUnsettledDeliveries.java

```diff
@@ -245,12 +280,12 @@ public Mono<Void> terminateAndAwaitForDispositionsInProgressToComplete() {
         //    or disposition requests
         isTerminated.getAndSet(true);
```

Comment:
If isTerminated is `true`, do we still want to continue with the logic down there? or is returning okay?

Reply:
similarly with the logic in `close()` line 326.

Reply:
`isTerminated` is set to true mainly for `onDelivery` (which can be called by broker) and `sendDisposition` (which can be called by user) to check for the state in case they are called post initiating the disposal, so it can emit error to caller.

ReceiverUnsettledDeliveries (and ReceiveLinkHandler2) goes through two stages when its being disposed of – (an optional) stage1 `preClose` and stage2 `close`.  The `terminateAndAwaitForDispositionsInProgressToComplete` part of `preClose`.  So, `isTerminated` not necessarily indicates both stages are triggered but mainly helps with check in `onDelivery` and `sendDisposition`.


The top-level call site (ReactorReceiver) that triggers the staged disposal will ensure that "closure happens only once" (through isClosedMono).